src/registrar/models/domain.py

```diff
@@ -267,25 +373,234 @@ def administrative_contact(self) -> PublicContact:
+            )
+            # TODO - ticket 433 human readable error handling here
+
+    def _update_domain_with_contact(self, contact: PublicContact, rem=False):
+        """adds or removes a contact from a domain"""
+        logger.info(
+            "_update_domain_with_contact() received type %s " % contact.contact_type
+        )
+        domainContact = epp.DomainContact(
+            contact=contact.registry_id, type=contact.contact_type
+        )
+
+        updateDomain = commands.UpdateDomain(name=self.name, add=[domainContact])
+        if rem:
+            updateDomain = commands.UpdateDomain(name=self.name, rem=[domainContact])
+
+        try:
+            registry.send(updateDomain, cleaned=True)
+        except RegistryError as e:
+            logger.error(
+                "Error changing contact on a domain. Error code is %s error was %s"
+                % (e.code, e)
+            )
+            action = "add"
+            if rem:
+                action = "remove"
+
+            raise Exception(
+                "Can't %s the contact of type %s" % (action, contact.contact_type)
+            )
+
+    @Cache
+    def security_contact(self) -> PublicContact:
+        """Get or set the security contact for this domain."""
+
+        # get the contacts: call _get_property(contacts=True)
+        # if contacts exist and security contact is in the contact list
+        # return that contact
+        # else call the setter
+        #   send the public default contact
+        try:
+            contacts = self._get_property("contacts")
+            for contact in contacts:
+                ##zander don't do this just to do the bare bones here
+                if "type" in contact.keys() and contact["type"] == PublicContact.ContactTypeChoices.SECURITY:
+                    tempContact= self.get_default_security_contact()
+                    tempContact.email=contact["email"]
+                    return tempContact
+
+        except Exception as err:  # use better error handling
+            logger.info("Couldn't get contact %s" % err)
+
+            # TODO - remove this! ideally it should return None,
+            # but error handling needs to be
+            # added on the security email page so that it can handle it being none
+        return self.get_default_security_contact()
+
+    def _add_registrant_to_existing_domain(self, contact: PublicContact):
+        self._update_epp_contact(contact=contact)
+
+        updateDomain = commands.UpdateDomain(
+            name=self.name, registrant=contact.registry_id
+        )
+        try:
+            registry.send(updateDomain, cleaned=True)
+        except RegistryError as e:
+            logger.error(
+                "Error changing to new registrant error code is %s, error is %s"
+                % (e.code, e)
+            )
+            # TODO-error handling better here?
+
+    def _set_singleton_contact(self, contact: PublicContact, expectedType: str):  # noqa
```

Comment:
In the function description, I'd recommend specifying what expectedType is used for in this context (an example would be useful here)